### PR TITLE
Nerfs janibuckets, buffs Blue-space beakers, rebalances its availablity

### PR DESCRIPTION
**Pull request by Nihisohel**
# About the pull request

Title. 
Done at a maintainers approval.

# Explain why it's good for the game

Its pretty ridiculous seeing all these loadouts requiring the usage of these very readily available buckets, and especially with a very high density of chemicals

This PR changes that by removing the fabricator craft for the janitorial bucket and makes it a large item, making it unable to stash it in your inventory, and nerfed its volume from 500u to 300u

Additionally, high-cap/bluespace beakers have been buffed to hold 500u from 300u, making it the more ideal item for chemical use, however its medilathe costs are increased from 10000 to 30000

As a little treat for those that will get angry, OT and Research gets one extra high-cap/bluespace beaker each within their respective offices

# Testing Photographs and Procedure
It works, find the beakers yourselves

# Changelog

:cl:
fix: Properly adds the smaller mop bucket to the autholathe
balance: Janitorial buckets are now considered large items, volume reduced from 500u to 300u
balance: High-cap beaker medilathe costs are increased from 10000 to 30000, volume increased from 300u to 500u
balance: OT has had their buckets replaced with high-cap beakers and OT and research gets one extra high-cap beakers within their respective offices
/:cl:

### CODE REVIEW

**Comment by Diegoflores31** [comment]
OT depends on janni buckets to make their work , cant you just leave jani buckets at 500 + bulky ??

File: code/game/objects/items/reagent_containers/glass.dm [hunk 1]
```diff
@@ -604,7 +604,7 @@
 
 
 /obj/item/reagent_container/glass/bucket
-	desc = "It's a bucket. Holds 120 units."
+	desc = "It's a bucket. Holds 100 units."
```
Comment (Diegoflores31) [inline code/game/objects/items/reagent_containers/glass.dm]: this is suposed to be 120 because the transfer amounts are 30 , 60 . this will break OT casings

File: code/game/objects/items/reagent_containers/glass.dm [hunk 2]
```diff
@@ -617,7 +617,7 @@
 	w_class = SIZE_MEDIUM
 	amount_per_transfer_from_this = 20
 	possible_transfer_amounts = list(10,20,30,60,120)
-	volume = 120
+	volume = 100
```
Comment (Diegoflores31) [inline code/game/objects/items/reagent_containers/glass.dm]: same as above

File: code/game/objects/items/reagent_containers/glass.dm [hunk 3]
```diff
@@ -679,23 +679,24 @@
...
-	desc = "A larger bucket, typically used with a mop. Holds 240 units"
+	desc = "A larger bucket, typically used with a mop. Holds 200 units"
 	icon_state = "mopbucket"
 	item_state = "mopbucket"
 	matter = list("metal" = 4000)
 	w_class = SIZE_LARGE
 	amount_per_transfer_from_this = 20
 	possible_transfer_amounts = list(10,20,30,60,120,240)
-	volume = 240
+	volume = 200
```
Comment (Diegoflores31) [inline code/game/objects/items/reagent_containers/glass.dm]: same as above

**Comment by BKryth** [comment]
> there is no reason to reduce the volume of those containers. nor have you explained why its being changed and pretty much breaks OT

OT casings dont require buckets and can work with just glass beakers and bottles. Though I will say that this is still an issue due to issues with floating error that happens when you mix at a weird transfer rate, in my experience Ive never had floating errors when mixing at 10/20/40/60 units, but occasionally do at anything else. I have no idea how to consistently replicate the floating error, it happens randomly.

Biggest problem I can see this with OT is when making paraformaldehyde with buckets/mopbuckets and it causing floating errors.

**Comment by Diegoflores31** [comment]
> > there is no reason to reduce the volume of those containers. nor have you explained why its being changed and pretty much breaks OT
> 
> OT casings dont require buckets and can work with just glass beakers and bottles. Though I will say that this is still an issue due to issues with floating error that happens when you mix at a weird transfer rate, in my experience Ive never had floating errors when mixing at 10/20/40/60 units, but occasionally do at anything else. I have no idea how to consistently replicate the floating error, it happens randomly.
> 
> Biggest problem I can see this with OT is when making paraformaldehyde with buckets/mopbuckets and it causing floating errors.

OTs quite literally spawn with buckets on their vendor and office. you are not expected to print on million beakers . plus as i said before there i no reason at all for the change

**Comment by DarkLordCabbage** [comment]
Oh hey, I remember floating the idea for this! (Thanks for doing all the work for me lol). I think the change going smoothly is contingent on bluespace beakers being readily available the way Jani buckets are for the departments that need them. I suggest just swapping out the jani buckets in the OT's office for the beakers in a 1:1 ratio and making even more available for research given that said department usually pilfers several from maints. 

Lowering the amount of 500u containers those departments have access to is an unintentional nerf and I think might meet with some resistance.

**Comment by halander1** [comment]
Makes the loss of beakers to acid smoke absolute cancer. As now they are extremely hard to replace. 

Which really comes back to the issue that you are balancing by making something more rare and more expensive. 

I really could care less if it is called a Jani bucket or a high cap breaker. What I do care about is a common supply of 500 I beakers for me to use in OT, med, and RnD.

That of which OT has none and most beg medsci for any. 

Med has a few but those are non-replacable. 

RnD has a few and can synth more. But now we expect beakers to cost glass to make instead of metal so I'm not making any more for anyone. 

At least make the beakers cost metal and move their recipe to the medilathe. Then give OT some. 

Overall just a severe negative QoL change.

**Comment by ihatethisengine** [comment]
Guys don't you think departments should not rely on JANITORIAL BUCKETS?

Even if we pretend you are -1ing this PR not because you want to keep your stupid minmax builds.

**Comment by Diegoflores31** [comment]
> Guys don't you think departments should not rely on JANITORIAL BUCKETS?
> 
> Even if we pretend you are -1ing this PR not because you want to keep your stupid minmax builds.

bro there is virtually no change to minmax builds . if that was the aim of the PR all that would need to be done is making buckets not fit on satchels.
now instead of pfcs with jannis you have pfcs with blue beaker ._.
there is no reason to just fuck with OT stuff

**Comment by sippykot** [comment]
> Guys don't you think departments should not rely on JANITORIAL BUCKETS?
> 
> Even if we pretend you are -1ing this PR not because you want to keep your stupid minmax builds.

i dont want coders fucking up the printable 500u yellow cauldrons of chemical goodness for no good reason

**Comment by Purplecorenuetral** [comment]
Seems like research is also going to deal with other departments asking for hicap beakers.

TL;DR: (Author of PR I wanna hear your thoughts) People are stealing glass, deconstructing windows for glass, piling up and banging on the doors of research every so often, distracting/slowing down researchers from their work now that research rework made stuff a bit more complicated.

The beakers cost 9 glass after this gets merged. There are very few unused glass in the almayer. If these unused-laying-around glass stacks runs out, people are going to be incentivised/forced to deconstruct shipside windows. You will see MTs/OTs, sometimes medical staff, deconstructing the nearest window, and pile infront of research's door asking for beakers, distracting them from preparing their research work.

And dont tell me to just "dont go to research then, cope with the beakers." People will inevitably ask research for beakers no matter what.

There are 3 known spawning glass stacks, not including engineering's glass:
OT spawns with 50.
Research spawns with 50.
Maintenance storage spawns 20 on the west side.
I didnt include engineering's glass because I dont wanna assumed to be correct on this.

Sure, you can buy glass with req... but they dont wanna spend budget just for 5-7 hi cap beakers to be printed that would likely be replacement for the melted/stolen/"randomly" disappeared prior beakers.

And dont tell me that "they should be careful next time", people make mistakes no matter what.

**Comment by Ammoniacres** [comment]
If the problem with janitorial buckets is either them being _goofy_ or tonally inconsistent with OT/Medical departments chemical use cases, or that they're too easily accessible to non-medical/OT personnel - then the solution should pretty easily be to create a new bucket that only those two groups can use, instead of punishing everyone (and especially research, who need the size of janitorial buckets quite often) for no particular reason.

**Comment by Purplecorenuetral** [comment]
> Guys don't you think departments should not rely on JANITORIAL BUCKETS?
> 
> Even if we pretend you are -1ing this PR not because you want to keep your stupid minmax builds.

TL;DR:
People are super used to the abundance of the 500 capacity janitorial bucket, it became the norm for beaker-chem-mixing. If you take it away immediately/without warning, do not expect a minor/small percent of the chem-based roles to react negatively. You cannot just say "stop using buckets and use this instead" where the "this" has a much more complicated way to "get" versus the target that was getting replaced.

This "min-max" build of chemical buckets is RARELY found being used, mostly due to the tedious process of procuring the chemicals, not-easy re-procuring them groundside and sacrificing a backpack slot/carrying around one. There is only one player I have heard of doing this consistently, they are not even 24/7 playing the game. Though I do agree that his loadout/builds mostly consist of chem buckets, and thats his reputation to deal with so I have no further say to the mentioned player.

Departments using janitorial buckets for mixing for like 2+ years is like an addiction at this point.

If you take away the abundance of janitorial buckets, lock them behind research, tripled price of hicap beakers, and combined with the already scarce source of glass and then deconstruction of windows for glass without proper CIC authorization, there might be an uproar to this.

You are basically taking away the staple/addictive "drug-like" nature of "buckets being seen and used as the base chem mixing beaker". It's like taking away candy from a baby that's been spoiled for 2 years with sweets, put the rest of the candies on the highest cabinet in the kitchen, and expecting no tantrum from the child.

I am planning on compiling both the discord's and this thread's suggested alternatives, but also add what would likely happen cons/pros. But I have no guarantee that I will post it here. No guarantees I'll post it here however.

**Comment by Diegoflores31** [comment]
> You are basically taking away the staple/addictive "drug-like" nature of "buckets being seen and used as the base chem mixing beaker". It's like taking away candy from a baby that's been spoiled for 2 years with sweets, put the rest of the candies on the highest cabinet in the kitchen, and expecting no tantrum from the child.

its not that people is spoiled , jani buckets like literallly exist so OT can mix their explosives on bulk . 
Chemlime can live with 300u beakers and they most of the time handle their stuff with that.
research on the other hand having a lathe on their shop are completely acustomed to having a janni bucket every single round.

this PR is unnecesary and all that needs to be done is to make jani buckets heavy so they dont fit on backpacks.

**Comment by Purplecorenuetral** [comment]
> its not that people is spoiled , jani buckets like literallly exist so OT can mix their explosives on bulk . Chemlime can live with 300u beakers and they most of the time handle their stuff with that. research on the other hand having a lathe on their shop are completely acustomed to having a janni bucket every single round.
> 
> this PR is unnecesary and all that needs to be done is to make jani buckets heavy so they dont fit on backpacks.

I stand corrected. I have only realized that "spoiled child" is not a good/accurate comparison of "people are treating buckets as the norm and its solidifying its position in the chemistry culture". The latter seems more accurate.

The only pros/neutrals I could find on the surface about this PR are:

High Capacity/Bluespace beakers living up to their name now.
Glass having 1 more more use other than window construction, medilathe, and OT explosive assemblies. (despite still being very limited in abundance-wise)
Research getting more customers about "Print [item] for me" (Gives them a bit more work diversity)
Xenos dont have to deal with a certain marine's chemical shenanigans that often anymore. (aka that guy doesnt exist anymore)
Medbay/Research doesn't have to deal with a certain marine's requests. (aka that guy doesnt exist anymore)

misc/neutral (consitives):
Chem bucket no longer chem-mixing meta, thus less printed and viewed as the pre-buff version of hicap beakers.
Chem bucket loadouts do not exist, thus driving people away from having a loadout pertaining to carry as much chemicals as possible. (unless an alternative was found, other wise this reduces loadout diversity/aspect of sandbox game/free will) (aka that certain chemical marine doesnt exist anymore)

**Comment by Diegoflores31** [comment]
> misc/neutral (consitives): Chem bucket no longer chem-mixing meta, thus less printed and viewed as the pre-buff version of hicap beakers. Chem bucket loadouts do not exist, thus driving people away from having a loadout pertaining to carry as much chemicals as possible. (unless an alternative was found, other wise this reduces loadout diversity/aspect of sandbox game/free will) (aka that certain chemical marine doesnt exist anymore)

i mean , from my perspective this PR changes nothing . it just reskins the janni buckets . players can still do the exact same tactic with the 1 extra step of just going research and ask for a beaker.

**Comment by Nihisohel** [comment]
> TL;DR: (Author of PR I wanna hear your thoughts) People are stealing glass, deconstructing windows for glass, piling up and banging on the doors of research every so often, distracting/slowing down researchers from their work now that research rework made stuff a bit more complicated.

If this gets researchers to interact with more people and have them fetch glass, through whatever means necessary, then all the more reason why this PR should get in, this is an RP server after all

If researchers go out of their way to give out their resources, then they should face the consequences of their actions by having a limited amount of said resources

Most of the current balancing was suggest by the maintainer, and I'm content on waiting for his review once a week passes to see if he changes his mind on some stuff

> i mean , from my perspective this PR changes nothing . it just reskins the janni buckets . players can still do the exact same tactic with the 1 extra step of just going research and ask for a beaker.

You are right when you say that it reskins the janibuckets, but it would be misleadling to simply title the PR as "Changes the sprite of the janibucket with the bluespace beaker" etc, etc, etc, when it changes other things as well

**Comment by sippykot** [comment]
>If this gets researchers to interact with more people and have them fetch glass, through whatever means necessary, then all the more reason why this PR should get in, this is an RP server after all

Researchers already have 999 things to do from botany to contracts to making chems for OT, stims manufacturing, xeno butchery... Having people line up at the door and scream "big beaker!!!" and run to smash every window on almayer for glass isn't rp.

Also, i just noticed that you are also the author of the UNGA nerf PR and all I have to say to that is... come on.

**Comment by Purplecorenuetral** [comment]
Over all, this PR got commented mostly negatively, likely people from departments that would obviously be the negative-effect victims. We need to see this in a Test Merge first if we are gonna be very sure about the predictions.

> If this gets researchers to interact with more people and have them fetch glass, through whatever means necessary, then all the more reason why this PR should get in, this is an RP server after all

I know giving roles more opportunities to roleplay is always good, especially that CM13 keeps the aspects of the roleplaying of SS13... but shoving it into the throat doesnt feel the same:
Making something that was generalized before, especially that it stayed that way for a very long time (in this case, janibuckets took the spot for being the best chem mixing beaker for at least 2 years, everyone is doing it and with no major complaints), doesnt end well if it wasnt executed in a good way. I have no examples to show you of such PR doing such thing/executing this kind of change properly, unfortunately.

But, we really do need to see this in a Test Merge if the negatives are negative, and the positives are positives. I recommend not speed merging this. Please just try it on a 4-days Test Merge. This PR affects many people, the crucial-to-the-operation/chem-maker-and-user people, even if its a small change; believe it or not, know it or not.

My wording may be "formal" or "nerdy" in which perspectives, and I may not be a contributor, but I am saying my thoughts as any other.

**Comment by Unknownity** [comment]
If the intention of the PR is to heavily cut down on PFCs taking high capacity beakers for their self use then the most obvious solution wouldn't be cutting the volume of the beakers themselves but changing only their size so they are unable to store them at all. (all the 120u+ beakers) Research, Medbay and OT would not be affected by size changes as they don't need to carry beakers around in their storage due to everything being in close proximity and just holding it in their hands is enough. This also would not affect how the departments operate. If your PR is aimed at marines, it should be done in a way that affects shipside the least.

At the current implementation, OT is heavily affected and will cause several issues in the department. Not because of jani bucket nerf or anything. But the unexplained nerfs of volume towards the regular and larger mop buckets, this will not only break certain aspects of OT production. It will also cause far more common instances of floating errors with those volume sizes. The 240 mop bucket's size is already large meaning marines can't store it already in their satchels/backpacks so nerfing its capacity seems redundant. The 120 mop bucket also being nerfed despite being a worse version of the 120 glass beaker (its size is larger than of the glass beaker) is also redundant and confusing. Who would take 120u mop buckets over the 120u glass beakers especially when storage would be a big issue for the marines? You rather should explain about this in "why its good for the game" because from our perspective it rather looks like you're intentionally nerfing OT when its not your stated goal.

File: code/game/objects/items/reagent_containers/glass.dm [hunk 4]
```diff
@@ -679,23 +679,24 @@
...
 /obj/item/reagent_container/glass/bucket/janibucket
 	name = "janitorial bucket"
-	desc = "It's a large bucket that fits in a janitorial cart. Holds 500 units."
+	desc = "It's a large bucket that fits in a janitorial cart. Holds 300 units."
 	icon_state = "janibucket"
 	item_state = "janibucket"
 	matter = list("metal" = 8000)
-	volume = 500
+	volume = 300
+	w_class = SIZE_LARGE
```
Comment (Ultimate-Fluff) [inline code/game/objects/items/reagent_containers/glass.dm]:
```suggestion
possible_transfer_amounts = list(5,10,15,20,25,30,40,50,60,80,100,120,150,240,300)

```
You updated the volume, so you need to update the transfer amounts to fit the volume of the container.

File: code/game/objects/items/reagent_containers/glass.dm [hunk 5]
```diff
@@ -679,23 +679,24 @@
...
 /obj/item/reagent_container/glass/bucket/mopbucket
 	name = "mop bucket"
-	desc = "A larger bucket, typically used with a mop. Holds 240 units"
+	desc = "A larger bucket, typically used with a mop. Holds 200 units"
 	icon_state = "mopbucket"
 	item_state = "mopbucket"
 	matter = list("metal" = 4000)
 	w_class = SIZE_LARGE
 	amount_per_transfer_from_this = 20
 	possible_transfer_amounts = list(10,20,30,60,120,240)
```
Comment (Ultimate-Fluff) [inline code/game/objects/items/reagent_containers/glass.dm]:
```suggestion
	possible_transfer_amounts = list(10,20,30,50,60,120,150,200)
```
You updated the volume, so you need to update the transfer amounts to fit the volume of the container.

File: code/game/objects/items/reagent_containers/glass.dm [hunk 6]
```diff
@@ -372,11 +372,11 @@
...
-	desc = "A beaker with an enlarged holding capacity, made with blue-tinted plexiglass in order to withstand greater pressure. Can hold up to 300 units."
+	desc = "A beaker with an enlarged holding capacity, made with blue-tinted plexiglass in order to withstand greater pressure. Can hold up to 500 units."
 	icon_state = "beakerbluespace"
 	item_state = "beakerbluespace"
-	matter = list("glass" = 10000)
-	volume = 300
+	matter = list("glass" = 30000)
+	volume = 500
 	amount_per_transfer_from_this = 10
 	possible_transfer_amounts = list(5,10,15,20,25,30,40,50,60,80,100,120,150,240,300)
```
Comment (Ultimate-Fluff) [inline code/game/objects/items/reagent_containers/glass.dm]:
```suggestion
	possible_transfer_amounts = list(5,10,15,20,25,30,40,50,60,80,100,120,150,240,250,300,400,500)
```
You updated the volume, so you need to update the transfer amounts to fit the volume of the container.

File: code/game/objects/items/reagent_containers/glass.dm (hunk 2 above)
Comment (kiVts) [inline code/game/objects/items/reagent_containers/glass.dm]: Revert this

File: code/game/objects/items/reagent_containers/glass.dm (hunk 3 above)
Comment (kiVts) [inline code/game/objects/items/reagent_containers/glass.dm]: Revert this

File: code/game/machinery/autolathe_datums.dm [hunk 7]
```diff
@@ -17,11 +17,6 @@
 	path = /obj/item/reagent_container/glass/bucket/mopbucket
 	category = AUTOLATHE_CATEGORY_GENERAL
 
-/datum/autolathe/recipe/mopbucket
-	name = "janitorial bucket"
-	path = /obj/item/reagent_container/glass/bucket/janibucket
-	category = AUTOLATHE_CATEGORY_GENERAL
-
```
Comment (kiVts) [inline code/game/machinery/autolathe_datums.dm:24]: readd the recipe

**Review by kiVts — CHANGES_REQUESTED** [review]
Replace all jani buckets in OT with space beakers
